app/assets/javascripts/modules/bucket-name.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+moj.Modules.bucketName = {
+  inputName: 'new-datasource-name',
+
+  init: function() {
+    var self = this;
+
+    if ($('#' + self.inputName).length) {
+      self.bindEvents();
+    }
+  },
+
+  bindEvents: function() {
+    var self = this,
+      $input = $('#' + self.inputName),
+      prefix = $input.data('prefix');
+
+    $input.on('keypress blur', function() {
+      self.checkValue($input, prefix);
+    });
+  },
+
+  checkValue: function($input, prefix) {
```

Comment:
Could `init` store the JQuery object for the input in the module, so that it doesn't need to be passed in to this function? This function could access `self.$input`. Or would that leak memory? The reference to the JQuery object is currently stored in the event handler closure anyway. Eg:
```javascript
init: function() {
  var self = this;
  self.$input = $('#' + self.inputName);
  if (self.$input.length) {
    self.prefix = self.$input.data('prefix');
    self.bindEvents();
  }
},

bindEvents: function () {
  var self = this;
  self.$input.on('keypress blur', function () {
    self.checkValue();
  });
},

checkValue: function () {
  var self = this;
  var val = self.$input.val();

  if (val.length < self.prefix.length) {
    ...
  }
  ...
}

Reply:
👍  on both points. Is it better practice to attach these things to the parent scope than to pass them around, then?